src/js/rtc_session.js

```diff
@@ -119,14 +121,14 @@ export class GrabLocalMediaState extends RTCSessionState {
         return "GrabLocalMediaState";
     }
     _gUM(constraints) {
-        return navigator.mediaDevices.getUserMedia(constraints);
+        return this._rtcSession._strategy._gUM(constraints);
     }
 }
 export class CreateOfferState extends RTCSessionState {
     onEnter() {
         var self = this;
         var stream = self._rtcSession._localStream;
-        self._rtcSession._pc.addStream(stream);
+        self._rtcSession._strategy.createOfferStateOnEnter(self._rtcSession._pc, stream);
```

Comment:
This function naming is very confusing - we are already inside OnEnter method and are calling onEnter again on the strategy. We might need to name this better

Reply:
Renamed